courses/views_test.py

```diff
@@ -239,3 +249,87 @@ def test_user_exists(self):
+
+# pylint: disable=no-self-use
+class CCXCreateTests(ApiTests):
+    """Test CCX create API for making CCXs on edx"""
+
+    def setUp(self):
+        self.payload = {
+            'master_course_id': 'foo',
+            'user_email': 'bleh@example.com',
+            'total_seats': 102,
+            'display_name': 'test name',
+        }
+        super(CCXCreateTests, self).setUp()
+
+    def test_missing_arguments_throw_error(self):
+        """Throw an error if we're missing a POST parameter"""
+        for key in ['master_course_id', 'user_email', 'total_seats', 'display_name']:
+            payload = self.payload.copy()
+            del payload[key]
+
+            result = self.client.post(reverse('create-ccx'), payload)
+
+            assert result.status_code == 400, str(result.content)
+            assert re.search('POST argument {}'.format(key).encode(), result.content)
+
+    def test_unknown_course_throws_404(self):
+        """If we're given an unknown master_course_id, throw a 404"""
+        result = self.client.post(reverse('create-ccx'), self.payload)
+
+        assert result.status_code == 404, str(result.content)
+
+    def test_request_error_returns_error(self):
+        """If there's an error with the edx request, return an error"""
+        course = CourseFactory.create()
+        self.payload['master_course_id'] = course.course_id
+
+        with mock.patch('courses.views.requests', autospec=True) as mock_req:
+            mock_req.post.side_effect = RequestException
+
+            result = self.client.post(reverse('create-ccx'), self.payload)
+
+            assert result.status_code == 502, str(result.content)
+
+    def test_errory_status_code_returns_error(self):
+        """If we get a status code from edx that looks error-y, throw an error"""
+        course = CourseFactory.create()
+        self.payload['master_course_id'] = course.course_id
+
+        with mock.patch('courses.views.requests', autospec=True) as mock_req:
+            mock_req.post.return_value.status_code = 500
+            mock_req.post.return_value.content = 'some error text'
+
+            result = self.client.post(reverse('create-ccx'), self.payload)
+
+            assert result.status_code == 502, str(result.content)
+
+    def test_201_on_happy_case(self):
+        """If all goes well, return 201"""
+        course = CourseFactory.create()
+        user_email = "john@example.com"
+        seats = 123
+        name = "CCX example title"
+        api_response = {
+            "ccx_course_id": "ccx-v1:Organization+EX101+RUN-FALL2099+ccx@1",
+            "display_name": name,
+            "coach_email": user_email,
+            "start": "2019-01-01",
+            "due": "2019-06-01",
+            "max_students_allowed": seats
+        }
+
+        with mock.patch('courses.views.requests', autospec=True) as mock_req:
+            mock_req.post.return_value.status_code = 201
+            mock_req.post.return_value.content = lambda: str(api_response)
```

Comment:
`str(api_response)` won't return valid JSON, we have to use `json.dumps` instead. But it doesn't appear that we use the return response at all, so why include it here?